modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp

```diff
@@ -871,134 +802,113 @@ void RenderableFieldlinesSequence::extractTriggerTimesFromFileNames() {
-bool RenderableFieldlinesSequence::extractCdfInfoFromDictionary(std::string& seedFilePath,
-                                                                std::string& tracingVar,
-                                                      std::vector<std::string>& extraVars)
-{
-    if (_dictionary->hasValue<std::string>(KeyCdfSeedPointFile)) {
-        seedFilePath = _dictionary->value<std::string>(KeyCdfSeedPointFile);
-        if (std::filesystem::is_regular_file(seedFilePath)) {
-            seedFilePath = absPath(seedFilePath).string();
-        }
-        else {
-            LERROR(fmt::format(
-                "{}: The specified seed poitn file: '{}' does not exist",
-                _identifier, seedFilePath
-            ));
-            return false;
-        }
+std::unordered_map<std::string, std::vector<glm::vec3>>
+                              RenderableFieldlinesSequence::extractSeedPointsFromFiles() {
+    std::vector<std::string> files;
+    std::filesystem::path seedPointDir;
+    std::unordered_map<std::string, std::vector<glm::vec3>> outMap;
+    
+    if (std::filesystem::is_directory(_seedPointDirectory)){
+        seedPointDir = absPath(_seedPointDirectory);
+        //_seedPointDirectory = seedPointDir.string();
     }
     else {
-        LERROR(fmt::format("{}: Must specify '{}'", _identifier, KeyCdfSeedPointFile));
-        return false;
-    }
-
-    if (_dictionary->hasValue<std::string>(KeyCdfTracingVariable)) {
-        tracingVar = _dictionary->value<std::string>(KeyCdfTracingVariable);
-    }
-    else {
-        tracingVar = "b"; //  Magnetic field variable as default
-        LWARNING(fmt::format(
-            "{}: No '{}', using default '{}'",
-            _identifier, KeyCdfTracingVariable, tracingVar
+        LERROR(fmt::format(
+            "The specified seed point directory: '{}' does not exist",
+            _seedPointDirectory
         ));
+        return outMap;
     }
 
-    if (_dictionary->hasValue<ghoul::Dictionary>(KeyCdfExtraVariables)) {
-        ghoul::Dictionary extraQuantityNamesDictionary =
-            _dictionary->value<ghoul::Dictionary>(KeyCdfExtraVariables);
-        const size_t nProvidedExtras = extraQuantityNamesDictionary.size();
-        for (size_t i = 1; i <= nProvidedExtras; ++i) {
-            extraVars.push_back(
-                extraQuantityNamesDictionary.value<std::string>(std::to_string(i))
-            );
+    namespace fs = std::filesystem;
+    for (const fs::directory_entry& spFile : fs::directory_iterator(seedPointDir)){
+        std::string seedFilePath = spFile.path().string();
+        if (!spFile.is_regular_file() || 
+                        seedFilePath.substr(seedFilePath.find_last_of('.')+1) != "txt") {
+            continue;
+        }
+        
+        std::ifstream seedFile(spFile);
+        if (!seedFile.good()) {
+            LERROR(fmt::format("Could not open seed points file '{}'", seedFilePath));
+            outMap.clear();
+            return outMap;
```

Comment:
No need to return the `outMap` if you cleared it immediately before;
`return {};` instead